src/getting_started/development/04_vscode_ext.md

```diff
@@ -0,0 +1,108 @@
+
+---
+
+## Contents
+- [SystemVerilog Language Server](#systemverilog-language-server)
+- [C++ Pack](#c-pack)
+- [Doxygen Generator](#doxygen-generator)
+- [Spell Checker](#spell-checker)
+- [Git & GitHub](#git--github)
+  - [GitLens](#gitlens)
+  - [GitHub Actions](#github-actions)
+  - [GitHub Pull Requests](#github-pull-requests)
+- [Markdown]()
+  - [Markdown All in One](#markdown-all-in-one)
+  - [Better Comments](#better-comments)
+- [Optional Bonuses](#optional-bonuses)
+  - [ColdCode](#coldcode)
+  - [Makefile Tools](#makefile-tools)
+  - [GitHub Style Markdown Preview](#github-style-markdown-preview)
+  - [Marp](#marp)
+  - [Rewrap](#rewrap)
+
+---
+
+### SystemVerilog Language Server
+
+[SVLangserver](https://marketplace.visualstudio.com/items?itemName=IMCTradingBV.svlangserver) is a language server for System Verilog that features auto-completion, module hierarchy, code snippets for common blocks, code formatting, syntax highlighting, and more.
+
+To install, find svlangserver in the VSCode marketplace. Then create and update a settings file at <WORKSPACE PATH>/.vscode/settings.json with the desired settings as detailed in the link above. The extension's commands and troubleshooting can also be found in the link.
+
+---
+
+### C++ Pack
+
+[This extension pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools-extension-pack) has 4 sets of popular extensions:
+  - [C/C++ support](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
+  - [C/C++ Themes](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools-themes)
+  - [CMake support](https://marketplace.visualstudio.com/items?itemName=twxs.cmake)
+  - [CMake Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools)
+  
+---
+  
+### Doxygen Generator
+  
+[Doxygen](https://github.com/doxygen/doxygen) is a documentation generator that extracts documentation from C++ source code comments. This [Doxygen comment generator](https://marketplace.visualstudio.com/items?itemName=cschlosser.doxdocgen) creates Doxygen comments for a variety of code snippets: constructors, functions, file descriptions, etc.
+  
+---
+
+### Spell Checker
+
+This [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) does exactly that: it is a basic spell check that can be customized and has a variety of add-on language dictionaries.
+
+---
+
+## Git & GitHub
+
+### GitLens
+  
+[GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) is a customizable extension that includes a wide variety of features such as a Git command palette, and detailed commit, repo, and branch histories. Full details can be found in the link above.
```

Comment:
It's worth pointing out that we _highly_ reccomend this and the pro version is free with github's student dev pack.